File: Java/JDI/jdi-uitest-core/src/main/java/com/epam/jdi/uitests/core/interfaces/common/ICheckBox.java
package com.epam.jdi.uitests.core.interfaces.common;
/*
 * Copyright 2004-2016 EPAM Systems
 *
 * This file is part of JDI project.
 *
 * JDI is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * JDI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; 
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with JDI. If not, see <http://www.gnu.org/licenses/>.
 */

import com.epam.jdi.uitests.core.interfaces.base.IClickable;
import com.epam.jdi.uitests.core.interfaces.base.ISetValue;
import ru.yandex.qatools.allure.annotations.Step;

/**
 * Created by Roman_Iovlev on 7/6/2015.
 */
public interface ICheckBox extends IClickable, IText, ISetValue {
    /**
     * Set checkbox checked
     */
    @Step
    void check();

    /**
     * Set checkbox unchecked
     */
    @Step
    void uncheck();

    /**
     * @return Verify is checkbox checked
     */
    @Step
    boolean isChecked();
}

File: C#.Net/JDI/Core/Interfaces/Common/ICheckBox.cs
using Epam.JDI.Core.Interfaces.Base;

namespace Epam.JDI.Core.Interfaces.Common
{
    public interface ICheckBox : IClickable, ISetValue
    {
        /**
         * Set checkbox checked
         */
        //TODO[JDIAction]
        void Check();

        /**
         * Set checkbox unchecked
         */
        //TODO[JDIAction]
        void Uncheck();

        /**
         * return: Verify is checkbox checked
         */
        //TODO[JDIAction]
        bool IsChecked();
    }
}
